types/puppeteer/index.d.ts

```diff
@@ -99,7 +110,7 @@ export interface EmulateOptions {
   userAgent?: string;
 }
 
-export type EvaluateFn<T> = (elem?: ElementHandle) => Promise<T>;
+export type EvaluateFn = string | ((...args: any[]) => any);
```

Comment:
Was the previous type too strict?

Reply:
Yes, the `elem?: ElementHandle` part is incorrect.

![image](https://user-images.githubusercontent.com/404765/31690263-d6c3b93a-b357-11e7-9180-d52fb4b5351d.png)

The `args` that will be passed to `pageFunction` are of type `Serializable|ElementHandle`.

`Serializable` is an `any` and the previous version locked it to only `ElementHandle`

It also returns a `Serializable` / `any`

https://github.com/GoogleChrome/puppeteer/blob/v0.12.0/docs/api.md#pageevaluatepagefunction-args